services/src/modules/authentication/strategies/jwt.ts

```diff
@@ -1,14 +1,11 @@
-import fastify from 'fastify';
-import { decode } from 'jsonwebtoken';
+import * as fastify from 'fastify';
 import logger from '../../logger';
 import { authenticationConfig } from '../../config';
 
 export default async function (request: fastify.FastifyRequest): Promise<void> {
-  const authHeader: string = request.headers.authorization;
-  if (!authHeader) throw new Error('Unauthorized');
-  const token = authHeader.split(' ')[1];
-  const decodedToken = decode(token, { json: true }) as Record<string, unknown>;
-  const issuer = String(decodedToken?.iss);
+  await request.decodeJWT();
```

Comment:
I don't really like the fact that we have to know this place is the first to use the JWT and manually call decode, and the other places just have to hope someone else already decoded it.

Two possible solutions:

1. Change `decodeJWT` function to first check if `decodedJWT` is already set, and do nothing if it is. Then, change every place that uses JWT to first call `decodeJWT` and then use it.

2. Turn `decodedJWT` into a proxy with a getter, where the getter actually does the decoding on the first invocation and simply returns the decoded token on the next invocations. Every place that needs to use `decodedJWT` simply uses it without needing to know if they are first or some place else previously decoded it.

option 2 is probably the better option, but takes a little more work

Reply:
Agree. I'll implement the first option for now.